mlir/include/mlir/TableGen/Class.h

```diff
@@ -521,6 +521,8 @@ class ParentClass {
   /// Write the parent class declaration.
   void writeTo(raw_indented_ostream &os) const;
 
+  friend class OpClass;
+
```

Comment:
Can you expose a `getTemplateParams` instead of using friend? Adding a friend in this case feels like a layering violation